omod-common/src/test/java/org/mockingbird/test/rest/resource/IdentifiedAnimal.java

```diff
@@ -0,0 +1,137 @@
+ * {@link org.openmrs.module.webservices.rest.web.OpenmrsClassScanner} and its classpath pattern.
+ * Necessary because existing fake test classes did not have the uuid needed to test
+ * fetching resources using the uuid.
+ */
+public class IdentifiedAnimal extends DelegatingCrudResource<Animal> {
+	private String name;
+	private String uuid;
+
+	/**
+	 * Instantiates a new Animal resource.
+	 * @param name the name
+	 * @param uuid the uuid
+	 */
+	public IdentifiedAnimal(String name, String uuid) {
+		this.name = name;
+		this.uuid = uuid;
+	}
+
+	/**
+	 * Gets name.
+	 * @return the name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Gets uuid.
+	 * @return the uuid
+	 */
+	public String getUuid() {
+		return uuid;
+	}
+
+	/**
+	 * Sets the name
+	 * @param name name
+	 */
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	/**
+	 * Sets the uuid
+	 * @param uuid uuid
+	 */
+	public void setUuid(String uuid) {
+		this.uuid = uuid;
+	}
+
+	/**
+	 * Gets the {@link DelegatingResourceDescription} for the given representation for this
+	 * resource, if it exists
+	 * @param rep representation
+	 * @return representation description
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		return null;
+	}
+
+	/**
+	 * Gets the delegate object with the given unique id. Implementations may decide whether
+	 * "unique id" means the uuid, or if they also want to retrieve delegates based on a unique
+	 * human-readable property.
+	 * @param uniqueId unique Id
+	 * @return the delegate for the given uniqueId
+	 */
+	@Override
+	public Animal getByUniqueId(String uniqueId) {
+		return null;
+	}
+
+	/**
+	 * Void or retire delegate, whichever action is appropriate for the resource type. Subclasses
+	 * need to override this method, which is called internally by
+	 * {@link #delete(String, String, RequestContext)}.
+	 * @param delegate animal object
+	 * @param reason reason
+	 * @param context context
+	 * @throws ResponseException exception
+	 */
+	@Override
+	protected void delete(Animal delegate, String reason, RequestContext context) throws ResponseException {}
+
+	/**
+	 * Purge delegate from persistent storage. Subclasses need to override this method, which is
+	 * called internally by {@link #purge(String, RequestContext)}.
+	 * @param delegate animal object
+	 * @param context context
+	 * @throws ResponseException exception
+	 */
+	@Override
+	public void purge(Animal delegate, RequestContext context) throws ResponseException {}
+
+	/**
+	 * Instantiates a new instance of the handled delegate
+	 * @return animal object
+	 */
+	@Override
+	public Animal newDelegate() {
+		return null;
+	}
+
+	/**
+	 * Writes the delegate to the database
+	 *
+	 * @param delegate animal object
+	 * @return the saved instance
+	 */
+	@Override
+	public Animal save(Animal delegate) {
+		return null;
```

Comment:
do we never need to save things here and always return a null? ,you can cross check the other methods too!!

Reply:
@HerbertYiga this is a fake resource class, it's used to mock a resource hence no need to persist anything. I used it as a template of a resource for which I can perform actions supported by the endpoints. You can have a look at the class 
JavaDoc comment.